internal/driver/common/canvas.go

```diff
@@ -560,13 +567,19 @@ type renderCacheTree struct {
 	root *RenderCacheNode
 }
 
-func updateLayout(objToLayout fyne.CanvasObject) {
+func (c *Canvas) updateLayout(objToLayout fyne.CanvasObject) {
 	switch cont := objToLayout.(type) {
 	case *fyne.Container:
 		if cont.Layout != nil {
-			cont.Layout.Layout(cont.Objects, cont.Size())
+			layout := cont.Layout
```

Comment:
Why do you need to lock to check `cont.Layout` but not `cont.Objects`?

Reply:
Indeed, good catch.